src/needUpdate.js

```diff
@@ -0,0 +1,23 @@
+import { parseString } from 'xml2js';
+import { fetchData } from './utils';
+
+export default (chromeStoreID, currentVersion) => new Promise((resolve, reject) =>
+  fetchData(
+    `https://clients2.google.com/service/update2/crx?x=id%3D${chromeStoreID}%26uc&prodversion=32`,
+  ).then((res) => {
+    if (res.statusCode === 200) {
+      parseString(res.body, (err, result) => {
+        const app = result.gupdate.app[0].$;
+        if (app.status !== 'ok') return resolve(false);
+
+        const { status, version: newestVersion } = result.gupdate.app[0].updatecheck[0].$;
+        if (status !== 'ok') return resolve(false);
+
+        if (newestVersion !== currentVersion) return resolve(true);
+        return resolve(false);
+      });
+    } else {
+      reject(`Failed to check current version of ${chromeStoreID}.`);
```

Comment:
Maybe we should log something and `resolve(true)` just in case?  Rejecting this promise will simply mess with developers, we can try to help them out as much as possible 👍